### ISSUE TITLE
Indexing bug in padding for unfrozen_elements

### ISSUE THREAD

**Issue by karthikviswanathn** [MeridianResearch/externalization#13]
There is a one-off indexing issue that happens between the teacher and a student. This can be observed in the [test_student_output.txt file](https://github.com/MeridianResearch/externalization/blob/02470797d0458f483e9de7fd2a0df609ff5c41c8/tests/test_student_output.txt). 

Summary:
- The input prompt length is 20 in the provided example, but the padding for `unfrozen_elements` in the `patched_forward` method is for 21 tokens.

Relevant code (method_patching.py):
```python
    elif self.early_exit_mode == 'sft_teacher':
        ...
        final_layer_logprobs = torch.stack(outputs.scores, 1).log_softmax(-1)[:,1:]       # [B, generation length - 1, vocab]
        assert self._early_exit_probabilities[0].cache == []
        gathered_early_exit_states = torch.concat([torch.stack(ee.cache, 1) for ee in self._early_exit_probabilities[1:]], 2)
        ...
```

It is likely related to the `[1:]` slicing in the above code, but this needs further investigation.
